cs3api4lab/api/cs3_file_api.py

```diff
@@ -139,67 +111,40 @@ def write_file(self, file_path, content, endpoint=None):
         Write a file using the given userid as access token. The entire content is written
         and any pre-existing file is deleted (or moved to the previous version if supported).
         """
-        #
-        # Prepare endpoint
-        #
-        time_start = time.time()
-        reference = file_utils.get_reference(file_path, endpoint)
-
-        if isinstance(content, str):
-            content_len = len(content)
-        else:
-            content_len = len(content.decode('utf-8'))
-        # providing '0' as size leads to unexpected additional file creation
-        content_size = str(content_len) if content_len > 0 else str(1)
-
-        meta_data = types.Opaque(map={"Upload-Length": types.OpaqueEntry(decoder="plain", value=str.encode(content_size))})
-        req = cs3sp.InitiateFileUploadRequest(ref=reference, opaque=meta_data)
-        init_file_upload_res = self.cs3_api.InitiateFileUpload(request=req, metadata=[
-            ('x-access-token', self.auth.authenticate())])
+        file_path = self.lock_manager.resolve_file_path(file_path, endpoint)
 
-        if init_file_upload_res.status.code != cs3code.CODE_OK:
-            self.log.debug('msg="Failed to initiateFileUpload on write" file_path="%s" reason="%s"' % \
-                           (file_path, init_file_upload_res.status.message))
-            raise IOError(init_file_upload_res.status.message)
+        time_start = time.time()
 
-        self.log.debug(
-            'msg="writefile: InitiateFileUploadRes returned" protocols="%s"' % init_file_upload_res.protocols)
+        content_size = FileUtils.calculate_content_size(content)
+        init_file_upload = self.storage_logic.init_file_upload(file_path, endpoint, content_size)
 
-        #
-        # Upload
-        #
         try:
-            protocol = [p for p in init_file_upload_res.protocols if p.protocol == "simple"][0]
-            headers = {
-                'Tus-Resumable': '1.0.0',
-                'File-Path': file_path,
-                'File-Size': content_size,
-                'x-access-token': self.auth.authenticate(),
-                'X-Reva-Transfer': protocol.token
-            }
-            put_res = requests.put(url=protocol.upload_endpoint, data=content, headers=headers)
+            upload_reponse = self.storage_logic.upload_content(file_path, content, content_size, init_file_upload)
 
+            self.lock_manager.lock_file(file_path, endpoint)
```

Comment:
Why are we creating a lock for a file_path that might be a conflict file? 
If it's a conflict file, can't we just fail this call with some different http code that the UI interprets? I mean, we save the conflict and return the error, so we don't do any locks.

Reply:
Yeah somehow I missed that